app/uk/gov/hmrc/merchandiseinbaggage/controllers/DeclarationController.scala

```diff
@@ -62,6 +62,24 @@ class DeclarationController @Inject()(declarationService: DeclarationService, mc
       )
   }
 
+  def findBy(mibReference: MibReference, eori: Eori): Action[AnyContent] = Action.async {
+    declarationService
+      .findBy(mibReference, eori)
+      .fold(
+        {
+          case DeclarationNotFound =>
+            logger.warn(s"Declaration not found for params: $mibReference, $eori")
+            NotFound
+          case e =>
+            logger.error(s"Error during findBy query - $e - params: $mibReference, $eori")
+            InternalServerError("Something went wrong")
+        },
+        foundDeclaration => {
+          Ok(toJson(FindByResponse(foundDeclaration.declarationId)))
+        }
+      )
+  }
+
```

Comment:
This will now need/benefit with contract test.....You'll need my branch to get merged. Once it get merged I could do for you...